pkg/virt-controller/watch/migration.go

```diff
@@ -448,10 +448,15 @@ func (c *MigrationController) updateStatus(migration *virtv1.VirtualMachineInsta
 		log.Log.Object(migration).Error("vmi's migration state was taken over by another migration object")
 	} else if vmi.Status.MigrationState != nil &&
 		vmi.Status.MigrationState.MigrationUID == migration.UID &&
-		vmi.Status.MigrationState.Failed {
+		(vmi.Status.MigrationState.Failed || vmi.Status.MigrationState.AbortRequested) {
```

Comment:
@vladikr Didn't we want to finalize the migration in the handler? This doesn't seems to be the case

Reply:
That is indeed true. I should communicate my thoughts better...

Virt-handler finalizes the migration in the `updateVMIStatus` function which comes after `vmUpdateHelperMigrationSource` where we cancel the migration.
This finalization is being done using the information stored in the domain metadata. 
If we would let virt-handler finalize this aborted migration, we would need to enhance the `updateVMIStatus` function so it would consider earlier updates to `vmi.Status.MigrationState.AbortStatus` and `.Failed` fields.
I'm worried that there may be some corner cases that will not clear these fields between migrations.

The currently proposed solution looked simple enough, so my comment was mainly to get your thoughts on it before implementing the said change in the virt-handler.

Reply:
I'm not sure that I'm following you guys :)

Currently, the flow looks something like this:
* Migration is being aborted (which essentially means that the vmim has a deletion timestamp)
* [virt-controller]: Reflects that fact in VMI's status
* [virt-handler]: Communicates with virt-launcher to cancel migration, then updates VMI's status when migration is abort succeeded.
* [virt-controller]: Marks the migration as failed and finally removed finalizer.

In my eyes there's a healthy separation of concerns here which aligns to the behavior that existed before this PR. virt-controller is responsible for the vmim object and target pods and virt-handler is responsible for the VMI and communicating with virt-launcher to actually cancel the migration. This separation fits very well IMO. Bear in mind that these two flows (dealing with vmim & target pods vs dealing the vmi & and virt-launcher) do not depend on each other and can occur in parallel.

Am I missing something here?

Reply:
The migration object gets finalized before the handler report successful abort. This is my concern @vladikr @iholder-redhat